packages/api/src/controllers/room.ts

```diff
@@ -64,14 +72,163 @@ function toExternalRoom(room: Room, isAdmin = false) {
   if (isAdmin) {
     return room;
   }
-  room.userId = undefined;
-  for (const key in room.participants) {
-    room.participants[key].tracksPublished = undefined;
+  const roomForResp = {
+    id: room.id,
+    updatedAt: room.updatedAt,
+    createdAt: room.createdAt,
+    participants: room.participants,
+  };
+
+  for (const key in roomForResp.participants) {
+    roomForResp.participants[key].tracksPublished = undefined;
   }
-  room.events = undefined;
-  return room;
+  return roomForResp;
 }
 
+app.delete("/:roomId", authorizer({}), async (req, res) => {
+  const room = await db.room.get(req.params.roomId);
+
+  if (!req.user.admin && req.user.id !== room.userId) {
+    throw new ForbiddenError(`users may only delete their own rooms`);
+  }
+
+  const svc = new RoomServiceClient(
+    req.config.livekitHost,
+    req.config.livekitApiKey,
+    req.config.livekitSecret
+  );
+  await svc.deleteRoom(req.params.roomId);
+
+  room.deleted = true;
+  room.deletedAt = new Date().getTime();
+  await db.room.replace(room);
+
+  res.status(204).end();
+});
+
+app.post(
+  "/:roomId/egress",
+  authorizer({}),
+  validatePost("room-egress-payload"),
+  async (req, res) => {
+    const room = await db.room.get(req.params.roomId);
+
+    if (!req.user.admin && req.user.id !== room.userId) {
+      throw new ForbiddenError(`users may only modify their own rooms`);
+    }
+
+    if (room.egressId !== undefined) {
+      throw new BadRequestError("egress already started");
+    }
+
+    const egressClient = new EgressClient(
+      req.config.livekitHost,
+      req.config.livekitApiKey,
+      req.config.livekitSecret
+    );
+    const output: StreamOutput = {
+      protocol: StreamProtocol.RTMP,
+      urls: [req.body.rtmpURL],
+    };
+
+    const info = await egressClient.startRoomCompositeEgress(
+      req.params.roomId,
+      output,
+      {
+        layout: "speaker-dark",
+        encodingOptions: {
+          keyFrameInterval: 2,
+        },
+      }
+    );
+    console.log("egress started", info);
+    room.egressId = info.egressId;
+    room.updatedAt = new Date().getTime();
+    await db.room.replace(room);
```

Comment:
Same regarding preference for `update` instead of full `replace`. These are just optional suggestions tho, will avoid making the same comment in other places below.